### PR TITLE
FIX : mine page no task found issue

### PR DESCRIPTION
**Pull request by sahsisunny**
### Closes: #748 

**Description of Changes:**
This pull request aims to address the issue of tasks not being displayed on the mine page in our platform. The problem is likely due to a bug in the tasks API, and this PR includes the necessary changes to rectify the situation.

**Changes Made:**

1. Updated Tasks API (RTK): The tasksAPI.ts file has been updated to ensure that it retrieves and provides the assigned tasks correctly to the mine page. The bug responsible for the missing tasks has been identified and fixed.
2. Mine Page Updates: The mine page components have been modified to handle the response from the updated tasks API and display the assigned tasks properly. Users should now be able to see their tasks on the mine page without any issues.

**Test Coverage:**
To ensure the correctness and robustness of the changes made, relevant tests have been added to cover the following scenarios:
1. tasksAPI.ts (RTK) Test: New test cases have been added to verify that the tasks API returns the correct tasks for a specific user.
tasksApi test coverage
2. Mine Page Test: The mine page component has been tested to ensure that it correctly renders tasks when there are assigned tasks and displays an appropriate message when no tasks are available for the user.
image

### Anything you would like to inform the reviewer about:
- No
### Dev Tested:
- [x] Yes

### Images/video of the change:

https://github.com/Real-Dev-Squad/website-status/assets/70854507/60f548b5-8da6-49ad-ba05-3c899e12ef64

### CODE REVIEW

**Comment by apurvkhare** [comment]
In the video when you click on mine, we briefly see 'not authorized, click to login'
Since you are already logged in that should not appear.
Is there a separate issue filed for it?
image

**Comment by sahsisunny** [comment]
> In the video when you click on mine, we briefly see 'not authorized, click to login' Since you are already logged in that should not appear. Is there a separate issue filed for it? image

This issue is related to [useAuthenticated](https://github.com/Real-Dev-Squad/website-status/blob/develop/src/hooks/useAuthenticated.ts) custom hook and @kotesh-arya is working on it (issue ticket #746 )

File: src/pages/mine/index.tsx [hunk 1]
```diff
@@ -30,7 +30,7 @@ const Content = () => {
 
     if (isLoading) return <p>Loading...</p>;
     if (error) return <p>Something went wrong! Please contact admin</p>;
-    if (tasks && tasks.tasks?.length)
+    if (tasks && tasks.length)
```
Comment (apurvkhare) [inline src/pages/mine/index.tsx:33]: I believe here just `tasks?.length` should work

Comment (sahsisunny) [reply src/pages/mine/index.tsx]: Yes its work

File: __mocks__/db/tasks.ts [hunk 2]
```diff
@@ -1,248 +1,312 @@
...
-    startedOn: '1618790400',
-    isNoteworthy: true,
-    title: 'Testing and Determinsitic State',
-    purpose: 'string',
-    percentCompleted: 0,
-    endsOn: '1618790400',
-    status: 'progress',
-    featureUrl: 'progress',
-    type: 'feature',
-    createdBy: 'shmbajaj',
```
Comment (gauravsinhaweb) [inline __mocks__/db/tasks.ts:25]: keep the file format as original, it is hard to review the changes (if any) in
`__mocks__/db/tasks.ts`

**Review by gauravsinhaweb — CHANGES_REQUESTED** [review]

File: __mocks__/db/tasks.ts (hunk 2 above)
Comment (sahsisunny) [reply __mocks__/db/tasks.ts:25]: The "prettier:fix" command rule does not seem to be applied to the mock database files. As a result, I had to manually fix the formatting of the data.

Comment (sahsisunny) [reply __mocks__/db/tasks.ts:25]: Please note that this is just dummy data, and there are no logic or code issues that would affect the application.

Comment (sahsisunny) [reply __mocks__/db/tasks.ts:25]: You can skip this file.

Comment (sahsisunny) [reply __mocks__/db/tasks.ts:25]: Test is passing right?

**Review by gauravsinhaweb — APPROVED** [review]

**Review by apurvkhare — APPROVED** [review]

**Review by hjaintech — APPROVED** [review]

**Review by Pratiyushkumar — APPROVED** [review]
LGTM

**Pull request merged**